ambari-common/src/main/python/resource_management/libraries/functions/mpack_version.py

```diff
@@ -0,0 +1,207 @@
+"""
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+"""
+
+import re
+
+"""
+ This class should be used to compare mpack and stack versions.
+ Base method which should be used is parse/parse_stack_version, depends
+ on which versions you want to compare. This method will validate and parse
+ version which you will pass as parameter, and return object of current class with
+ parsed version. Same thing you should do with another version, with which you are
+ planning to compare previous one. After that, use "==", ">", "<" to get final result.
+"""
+class MpackVersion(object):
+  __module_version_pattern = "(?P<major>[0-9]+).(?P<minor>[0-9]+).(?P<maint>[0-9]+)(-h(?P<hotfix>[0-9]+))*-b(?P<build>[0-9]+)"
```

Comment:
Hi,
If we want to support for eg HDP-2.6.1.0 then we will have __module_legacy_stack_version_pattern to match that and __mpack_version_pattern to match new schema 
<MAJOR>.<MINOR>.<MAINT>-b<BUILDNUM>
<MAJOR>.<MINOR>.<MAINT>-h<HOTFIXNUM>-b<BUILDNUM>
I am not sure I understand the need for module version pattern since you have a separate class to handle it.

Reply:
I believe this should be called mpack_version_pattern, mpack_legacy_stack_version_pattern, mpack_version_regex etc instead of "module"

Reply:
Done